src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -228,6 +238,18 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                     _traceWriter?.Error(message, ex);
                     _logger?.LogError(0, ex, message);
 
+                    if (!IsHostHealthy() && _healthCheckWindow.GetEvents(isHealthy => !isHealthy).Count() > _config.HostHealthMonitor.HealthCheckThreshold)
```

Comment:
I'd extract this into a method, maybe something like ShouldRestartScriptHost?

Reply:
Done